synapse/models/inet.py

```diff
@@ -1442,7 +1442,11 @@ def getModelDefs(self):
                         'doc': 'A channel within a web service or instance such as slack or discord.'
                     }),
 
-                    ('inet:web:hashtag', ('str', {'lower': True, 'regex': r'^#\w[\w·]*(?<!·)$'}), {
+                    ('inet:web:hashtag', ('str', {'lower': True, 'strip': True, 'regex': r'^#[^\p{Z}#]+$'}), {
+                        # regex explanation:
+                        # - starts with pound
+                        # - one or more non-whitespace/non-pound character
+                        # The minimum hashtag is a pound with a single non-whitespace character
```

Comment:
Any chance we could create a disallow list for some special syntax chars that we presume any sane implementation would disallow?

Reply:
That first set of characters in the square brackets basically is the disallow list (whitespace and `#`). Any other suggestions?